adam-core/src/main/scala/org/bdgenomics/adam/converters/AlignmentRecordConverter.scala

```diff
@@ -202,6 +200,64 @@ class AlignmentRecordConverter extends Serializable {
 
     samHeader
   }
+
+  /**
+   * Converts a fragment to a set of reads.
+   *
+   * @param fragment Fragment to convert.
+   * @return The collection of alignments described by the fragment. If the fragment
+   *         doesn't contain any alignments, this method will return one unaligned
+   *         AlignmentRecord per sequence in the Fragment.
+   */
+  def convertFragment(fragment: Fragment): Iterable[AlignmentRecord] = {
+    val sequences: List[Sequence] = asScalaBuffer(fragment.getSequences).toList
+    require(sequences.nonEmpty, "Fragment %s contains no sequences.".format(fragment))
+
+    // create fragment builders
+    val builders = if (fragment.getAlignments.isEmpty) {
+      List.tabulate(sequences.length)(id =>
+        AlignmentRecord.newBuilder()
+          .setSequence(sequences(id).getBases)
+          .setQual(sequences(id).getQualities)
+          .setReadNum(id))
+    } else {
+      val alignments: List[AlignmentRecord] = asScalaBuffer(fragment.getAlignments).toList
+      alignments.map(r => {
+        require(r.getReadNum != null, "Read number is null for read %s in fragment %s.".format(
+          r, fragment))
+        val (seq, qual) = if (r.getReadNegativeStrand) {
+          (Alphabet.dna.reverseComplement(sequences(r.getReadNum).getBases,
+            (c: Char) => Symbol('N', 'N')),
+            sequences(r.getReadNum).getQualities.reverse)
+        } else {
+          (sequences(r.getReadNum).getBases,
```

Comment:
Ah, the assumption I was making here was that when we had data stored as `Fragment`s, we would null out the sequence and quality inside of each nested `AlignmentRecord`. This would be slightly more efficient for multiply aligned reads.